app/src/main/java/com/mariogrip/octodroid/UpdateGraph.java

```diff
@@ -0,0 +1,199 @@
+        HotEndActTemp.setCircleColor(Color.rgb(127, 255, 0));
+        HotEndActTemp.setCircleSize(5f);
+        HotEndActTemp.setFillColor(Color.rgb(127, 255, 0));
+        HotEndActTemp.setDrawCubic(true);
+        HotEndActTemp.setDrawValues(false);
+
+
+       LineDataSet BedTargetTemp = new LineDataSet(BedTempTarList, "Bed Temp Target");
+        BedTargetTemp.setColor(Color.rgb(72, 61, 39));
+        BedTargetTemp.setLineWidth(2.5f);
+        BedTargetTemp.setCircleColor(Color.rgb(240, 238, 70));
+        BedTargetTemp.setCircleSize(5f);
+        BedTargetTemp.setFillColor(Color.rgb(240, 238, 70));
+        BedTargetTemp.setDrawCubic(true);
+        BedTargetTemp.setDrawValues(false);
+
+        LineDataSet HotEndTargetTemp = new LineDataSet(HotEndTempTarList, "Hot End Temp Target");
+        HotEndTargetTemp.setColor(Color.rgb(255, 69, 0));
+        HotEndTargetTemp.setLineWidth(2.5f);
+        HotEndTargetTemp.setCircleColor(Color.rgb(240, 238, 70));
+        HotEndTargetTemp.setFillColor(Color.rgb(240, 238, 70));
+        HotEndTargetTemp.setDrawCubic(true);
+        HotEndTargetTemp.setDrawValues(false);
+
+
+
+        BedActTemp.setAxisDependency(YAxis.AxisDependency.LEFT);
+
+        dataSets.add(BedActTemp);
+        dataSets.add(HotEndActTemp);
+        dataSets.add(BedTargetTemp);
+        dataSets.add(HotEndTargetTemp);
+
+
+
+        LineData data = new LineData(xVals, dataSets);
+        return data;
+
+    }
+
+    private static ArrayList<Entry> getTempArray(TempType type) {
+        ArrayList<Entry> tempHistory = new ArrayList<>();
+        ArrayList<HistoryObject> historyArrayList = memory.temp.history.getHistoryArray();
+
+        int interval = historyArrayList.size()/15;
+        Log.i(interval+"","");
+        switch (type){
+            case BedTarget:
+                for(int i =0; i < historyArrayList.size() && historyArrayList.size()!=0; i+= interval){
+                   tempHistory.add(new Entry(historyArrayList.get(i).getBedTargetTemp(),i));
+                }
+                break;
+            case BedActual:
+                for(int i =0; i < historyArrayList.size() && historyArrayList.size()!=0; i+=interval){
+                    tempHistory.add(new Entry(historyArrayList.get(i).getBedActTemp(), i));
+
+                }
+                break;
+            case HotEndActual:
+                for(int i =0; i < historyArrayList.size() && historyArrayList.size()!=0; i+= interval){
+                    tempHistory.add(new Entry(historyArrayList.get(i).getHotEndActTemp(),i));
+                }
+                break;
+            case HotEndTarget:
+                for(int i =0; i < historyArrayList.size() && historyArrayList.size()!=0; i+= interval){
+                    tempHistory.add(new Entry(historyArrayList.get(i).getHotEndTargetTemp(), i));
+                }
+                break;
+
+        }
+
+
+        return  tempHistory;
+    }
+
+    private static ArrayList<String> getXVals(int length) {
```

Comment:
This method creates the Xaxis values. The math to the test string value works for about 10 minutes and then appears to break (needs to be fixed in a future commit)